fedimint-core/src/db/mod.rs

```diff
@@ -286,6 +313,79 @@ pub trait IDatabaseTransaction<'a>: 'a + Send {
     async fn set_tx_savepoint(&mut self);
 }
 
+pub struct IDatabaseTransactionWrapper<'a, Tx: IDatabaseTransactionCore<'a> + Send>(
+    Option<Tx>,
+    &'a PhantomData<()>,
+);
+
+impl<'a, Tx: IDatabaseTransactionCore<'a> + Send> IDatabaseTransactionWrapper<'a, Tx> {
+    pub fn new(dbtx: Tx) -> IDatabaseTransactionWrapper<'a, Tx> {
+        IDatabaseTransactionWrapper(Some(dbtx), &PhantomData)
+    }
+}
+
+#[async_trait]
+impl<'a, Tx: IDatabaseTransactionCore<'a> + Send> IDatabaseTransaction<'a>
+    for IDatabaseTransactionWrapper<'a, Tx>
+{
+    async fn raw_insert_bytes(&mut self, key: &[u8], value: Vec<u8>) -> Result<Option<Vec<u8>>> {
+        match &mut self.0 {
+            Some(tx) => tx.raw_insert_bytes(key, value).await,
+            None => panic!("Cannot insert into already consumed transaction"),
```

Comment:
I wonder if it wouldn't be better to return an error instead...

Reply:
Changed to return error where possible.